downloader/src/main/kotlin/vcs/Git.kt

```diff
@@ -97,13 +98,17 @@ class Git : VersionControlSystem(), CommandLineTool {
 
     override val type = VcsType.GIT
     override val priority = 100
-    override val defaultBranchName = "master"
     override val latestRevisionNames = listOf("HEAD", "@")
 
     override fun command(workingDir: File?) = "git"
 
     override fun getVersion() = getVersion(null)
 
+    override fun getDefaultBranchName(url: String): String? {
+        val refs = Git.lsRemoteRepository().setRemote(url).callAsMap()
+        return (refs["HEAD"] as? SymbolicRef)?.target?.name?.removePrefix("refs/heads/")
```

Comment:
Would it make sense to provide a fallback value here, or is this expected to always return a value?

Reply:
Remote repos are indeed not obliged to contain a symbolic `HEAD` ref to denote the default branch. The question is, what should be the fallback value, "master" or "main"?

Reply:
For now probably "master" is the better choice as I expect that the majority of repos still use it. Maybe the best would be to implement the fallback in the calling code and try "master" first, and if it fails try also with "main". But I don't know if it's worth the effort because I have no idea how many repositories do not have `HEAD`.

Reply:
At least all GitHub report have a `HEAD` ref, and I barely came across repos that don't do I went with using "master" as a fallback to mostly closely resemble the original behavior.